File: backend/rest-api-db/rest-api-db/src/main/java/com/icici/rest_api_db/entities/Itinerary.java
package com.icici.rest_api_db.entities;

import java.time.LocalDate;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;

@Entity
@Data
public class Itinerary {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    int id;
    String title;
    String description;
    LocalDate date;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    List<Activity> activities;

    

    

    
}

File: backend-nodejs/node-rest-api-db/schemas/Itinerary.js
const mongoose = require('mongoose');

const ItinerarySchema = new mongoose.Schema({
    title: { type: String, required: true },
    description: String,
    date: Date,
    activities:  [{ type: mongoose.Schema.Types.ObjectId, ref: 'Activity' }]
});

const Itinerary = mongoose.model('Itinerary', ItinerarySchema);

module.exports = Itinerary;